earth_enterprise/src/fusion/dbmanifest/dbmanifest_unittest.cpp

```diff
@@ -145,21 +250,310 @@ TEST_F(DbManifestTest, EnforcesAbsolutePath) {
+  db_create_params.icon_files.push_back(khComposePath(assets_root_, "SomeDataDir/inconpath2/icon4.png"));
+  db_create_params.icon_files.push_back(khComposePath(assets_root_, "SomeDataDir/inconpath2/icon5.gnp"));  // negitive test
+
+  CreateMockDbFolderWithPoiData(db_create_params);
 
-  CreateFakeDbFolderWithPoiData();
+  // initialize mock get index manifest call
+  // create unitialized entries to avoid OS calls for file sizes that don't exist
+  mock_get_index_manifest->manifest_.resize(3);
+  // for testing purposes keep it simple and stupid.  No i/o is done to these files
+  mock_get_index_manifest->manifest_[0].orig_path = khComposePath(assets_root_, "SomeDataDir/somedatafile1");
+  mock_get_index_manifest->manifest_[0].current_path = mock_get_index_manifest->manifest_[0].orig_path;
+  mock_get_index_manifest->manifest_[0].data_size = 1024;
+  mock_get_index_manifest->manifest_[1].orig_path = khComposePath(assets_root_, "SomeDataDir/somedatafile2");
+  mock_get_index_manifest->manifest_[1].current_path = mock_get_index_manifest->manifest_[0].orig_path;
+  mock_get_index_manifest->manifest_[1].data_size = 1024;
+  mock_get_index_manifest->manifest_[2].orig_path = khComposePath(assets_root_, "SomeDataDir/somedatafile3");
+  mock_get_index_manifest->manifest_[2].current_path = mock_get_index_manifest->manifest_[0].orig_path;
+  mock_get_index_manifest->manifest_[2].data_size = 1024;
 
-  EXPECT_NO_THROW(db_manifest = TransferOwnership(new DbManifest(&db_path)));
+  expected_stream_manifest = mock_get_index_manifest->manifest_;
+  expected_stream_manifest.push_back(ManifestEntry(db_create_params.poi_entries[0].poi_file));
+  expected_search_manifest.push_back(ManifestEntry(db_create_params.poi_entries[0].poi_file));
+  expected_stream_manifest.back().dependents.push_back(ManifestEntry(db_create_params.poi_entries[0].poi_data_files[0]));
+  expected_search_manifest.back().dependents.push_back(ManifestEntry(db_create_params.poi_entries[0].poi_data_files[0]));
+  expected_stream_manifest.push_back(ManifestEntry(toc_path_));
+  expected_stream_manifest.push_back(ManifestEntry(db_header_path_));
+  expected_stream_manifest.push_back(ManifestEntry(db_create_params.icon_files[0]));
+  expected_stream_manifest.push_back(ManifestEntry(db_create_params.icon_files[1]));
+  expected_stream_manifest.push_back(ManifestEntry(db_create_params.icon_files[2]));
+  expected_stream_manifest.push_back(ManifestEntry(db_create_params.icon_files[3]));
+
+  // change pointer to base type
+  get_index_manifest = TransferOwnership(mock_get_index_manifest);
+
+  EXPECT_NO_THROW(db_manifest = TransferOwnership(new DbManifest(&db_path, get_index_manifest)));
   EXPECT_EQ(db_path_v1_, db_path); // should be unchanged
+  EXPECT_TRUE(db_manifest->Prefix().empty());
+
+  db_manifest->GetPushManifest(file_pool_, &stream_manifest, &search_manifest, "", "");
+
+  bool is_equivalent = IsEquivalent(&expected_stream_manifest, &stream_manifest);
+  if (!is_equivalent) {
```

Comment:
should this fail the test execution instead of a warning ?

Reply:
line 437 will fail the test.  To make failures easier to diagnose I added this pre-check and dump the manifests before ding the EXPECT_TRUE() below which will do the actual failing of the test.

Reply:
i see. its for diagnostic purposes.